packages/libs/error-stack/src/ext/iter.rs

```diff
@@ -0,0 +1,338 @@
+//! This module provides adaptors which make it possible to use `error-stack`'s functionality
+//! on [`Iterator`]s.
+//!
+//! Most of the relevant documentation can be found in [`IteratorExt`].
+
+use core::{
+    fmt::{Debug, Display},
+    marker::{Send, Sync},
+};
+
+use crate::{Context, Report, ResultExt};
+
+/// This trait provides extension methods for iterators which makes it possible to manipulate
+/// items inside them using the usual `error-stack` methods.
+///
+/// This trait is only implemented for iterators of items which implement [`crate::ResultExt`].
+///
+/// Because this trait attaches the provided data to _every_ item in the iterator, the
+/// provided data must implement [`Clone`].
```

Comment:
This is not true anymore as attaching it lazily does not require it to be clonable.